INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8333
 * Running on http://10.0.0.7:8333
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:root:GET request received for ModeS: 4ACA46
INFO:root:Aircraft found for ModeS: 4ACA46
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:39:20] "GET /aircraft/4ACA46 HTTP/1.1" 200 -
INFO:root:GET request received for Registration: SP-LVA
INFO:root:Aircraft found for Registration: SP-LVA
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:39:37] "GET /aircraft/registration/SP-LVA HTTP/1.1" 200 -
INFO:root:GET request received for ModeS: 48D4A0
INFO:root:Aircraft found for ModeS: 48D4A0
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:40:35] "GET /aircraft/48D4A0 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:43:58] "[33mPOST /aircraft/ HTTP/1.1[0m" 404 -
ERROR:main:Exception on /aircraft [POST]
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\default.py", line 924, in do_execute
    cursor.execute(statement, parameters)
sqlite3.IntegrityError: UNIQUE constraint failed: Aircraft.ModeS

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\Bruce\ADS-BPro_API\main.py", line 52, in add_aircraft
    db.session.commit()
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\scoping.py", line 597, in commit
    return self._proxied.commit()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 2017, in commit
    trans.commit(_to_root=True)
  File "<string>", line 2, in commit
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 1302, in commit
    self._prepare_impl()
  File "<string>", line 2, in _prepare_impl
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\state_changes.py", line 139, in _go
    ret_value = fn(self, *arg, **kw)
                ^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 1277, in _prepare_impl
    self.session.flush()
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 4341, in flush
    self._flush(objects)
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 4476, in _flush
    with util.safe_reraise():
  File "C:\Python312\Lib\site-packages\sqlalchemy\util\langhelpers.py", line 146, in __exit__
    raise exc_value.with_traceback(exc_tb)
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\session.py", line 4437, in _flush
    flush_context.execute()
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\unitofwork.py", line 466, in execute
    rec.execute(self)
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\unitofwork.py", line 642, in execute
    util.preloaded.orm_persistence.save_obj(
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\persistence.py", line 93, in save_obj
    _emit_insert_statements(
  File "C:\Python312\Lib\site-packages\sqlalchemy\orm\persistence.py", line 1233, in _emit_insert_statements
    result = connection.execute(
             ^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1418, in execute
    return meth(
           ^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\sql\elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1640, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 2353, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "C:\Python312\Lib\site-packages\sqlalchemy\engine\default.py", line 924, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) UNIQUE constraint failed: Aircraft.ModeS
[SQL: INSERT INTO aircraft ("FirstCreated", "LastModified", "ModeS", "ModeSCountry", "Country", "Registration", "Manufacturer", "ICAOTypeCode", "Type", "SerialNo", "YearBuilt", "PictureURL1", "PictureURL2", "PictureURL3", "OperatorFlagCode") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('2024-07-29 10:44:15.506141', '2024-07-29 10:44:15.506141', '48D980', 'Poland', 'Poland', '0110', 'Boeing', 'B738', 'Boeing 737-800WL', '61358/6370', '2017', 'https://cdn.radarbox.com/photo/0110-1720047601-0.jpg', None, None, 'PLF')]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:44:15] "[35m[1mPOST /aircraft HTTP/1.1[0m" 500 -
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:45:55] "[31m[1mPUT /aircraft/48D980 HTTP/1.1[0m" 405 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8333
 * Running on http://10.0.0.7:8333
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:46:29] "PUT /aircraft/48D980 HTTP/1.1" 200 -
INFO:root:GET request received for ModeS: 48D4A0
INFO:root:Aircraft found for ModeS: 48D4A0
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:48:13] "GET /aircraft/48D4A0 HTTP/1.1" 200 -
INFO:root:GET request received for Registration: SP-LVA
INFO:root:Aircraft found for Registration: SP-LVA
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:48:51] "GET /aircraft/registration/SP-LVA HTTP/1.1" 200 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8333
 * Running on http://10.0.0.7:8333
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:50:46] "[31m[1mPUT /aircraft/48D4A0 HTTP/1.1[0m" 405 -
INFO:werkzeug:[31m[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.[0m
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:8333
 * Running on http://10.0.0.7:8333
INFO:werkzeug:[33mPress CTRL+C to quit[0m
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:51:10] "PUT /aircraft/48D4A0 HTTP/1.1" 200 -
INFO:root:GET request received for ModeS: 48D4A0
INFO:root:Aircraft found for ModeS: 48D4A0
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:51:16] "GET /aircraft/48D4A0 HTTP/1.1" 200 -
INFO:root:GET request received for ModeS: 48D4A0
INFO:root:Aircraft found for ModeS: 48D4A0
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:52:21] "GET /aircraft/48D4A0 HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:56:39] "[33mGET / HTTP/1.1[0m" 404 -
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:56:39] "[33mGET /favicon.ico HTTP/1.1[0m" 404 -
INFO:root:GET request received for ModeS: 48DA4A
WARNING:root:Aircraft not found for ModeS: 48DA4A
INFO:werkzeug:127.0.0.1 - - [29/Jul/2024 10:56:44] "[33mGET /aircraft/48DA4A HTTP/1.1[0m" 404 -
